### ISSUE TITLE
Tunnel setup for sit-type tunnel fails

### ISSUE THREAD

**Issue by shartge** [systemd/systemd#2758]
### Submission type

[X] Bug report
### systemd version the issue has been seen with

> 229
### Used distribution

> Debian Sid
### In case of bug report: Expected behaviour you didn't see

> Working sit tunnel
### In case of bug report: Unexpected behaviour you saw

> Not working sit tunnel
### In case of bug report: Steps to reproduce the problem

I am using systemd-networkd to manage a HE.net IPv6 Tunnel, using a netdev file and two network files in /etc/systemd/network:                                                                                                                              

he-tunnel.netdev:

```
[Match]                                                                                                                                                                               

[NetDev]                                                                                                                                                                              
Description=HE.net Tunnelbroker sit device                                                                                                                                            
Name=he-ipv6                                                                                                                                                                          
Kind=sit                                                                                                                                                                              
MTUBytes=1480                                                                                                                                                                         

[Tunnel]                                                                                                                                                                              
Local=130.180.40.117                                                                                                                                                                  
Remote=216.66.80.30                                                                                                                                                                   
TTL=255                                     
```

he-tunnel.network:

```
[Match]                                                                                                                                                                               
Name=he-ipv6                                                                                                                                                                          

[Network]                                                                                                                                                                             
Address=2001:470:1f0a:cbf::2/64                                                                                                                                                       
Gateway=2001:470:1f0a:cbf::1                                                                                                                                                          

IPForward=true                          
```

lan.network:

```
[Match]                                                                                                                                                                               
MACAddress=02:17:08:81:64:44                                                                                                                                                          

[Network]                                                                                                                                                                             
Address=130.180.40.117/29                                                                                                                                                             
Gateway=130.180.40.113                                                                                                                                                                

Address=2001:470:7403:f::1/64                                                                                                                                                         

Tunnel=he-ipv6                                                                                                                                                                        

IPForward=true                                                                                                                                                                        

[Route]                                                                                                                                                                               
Destination=2001:470:7403::/48                                                                                                                                                        
Gateway=2001:470:7403:f::2                                                                                                                                                            

[Route]                                                                                                                                                                               
Destination=192.168.100.0/24                                                                                                                                                          
Gateway=130.180.40.114           
```

This has worked flawless so far, but after a reboot today, systemd-networkd does not want to create the tunnel device and logs the following error:                                                                                                               

```
Feb 27 20:18:47 urd systemd-networkd[1654]: [/etc/systemd/network/he-tunnel.netdev:10] Tunnel addresses incompatible, ignoring assignment: 130.180.40.117
Feb 27 20:18:47 urd systemd-networkd[1654]: [/etc/systemd/network/he-tunnel.netdev:11] Tunnel addresses incompatible, ignoring assignment: 216.66.80.30
Feb 27 20:18:47 urd systemd-networkd[1654]: Tunnel with invalid address family configured in /etc/systemd/network/he-tunnel.netdev. Ignoring
Feb 27 20:18:47 urd systemd-networkd[1654]: [/etc/systemd/network/lan.network:10] Tunnel is invalid, ignoring assignment: he-ipv6
```

It seems systemd-network expects an IPv6 address for local and remote which is of course totally wrong for a 6-in-4 tunnel.

If I manually create the tunnel interface he-ipv6 with

```
ip tunnel add he-ipv6 mode sit remote 216.66.80.30 local 130.180.40.117 ttl 255                                                                                                     
ip link set he-ipv6 up mtu 1480                    
```

then systemd-networkd uses the information in he-tunnel.network to add the IPs to the interface but it is not able to create the tunnel interface in the first place because it wrongly expects an IPv6-Adresse in the local and remote options.

This is also [Debian Bug#816132](http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=816132).

**Comment by marckleinebudde** [comment]
Looks like a duplicate of #2655 - which was fixed with # #2692.

**Comment by shartge** [comment]
Looks like it. I am building a patched version right now to verify.

**Comment by shartge** [comment]
Yes, the fix from #2692 also fixes this issue.

**Comment by marckleinebudde** [comment]
If the issue is fixed, I think you can close it.
